ingest_python_scripts/prechecks.py

```
@@ -0,0 +1,1118 @@
+        return True
+    if "<error" in lower:
+        bad("OAI endpoint returned an <error> element. Review output above.")
+        return False
+    # Detect HTML error pages (403, 404, 502, etc.) — curl exits 0 but the
+    # response is an nginx/Apache error page, not OAI XML. Non-blocking caution
+    # since some endpoints block the precheck IP but still allow the harvester.
+    http_error = re.search(r"<title>\s*(\d{3})[^<]*</title>", body, re.IGNORECASE)
+    if http_error:
+        code = http_error.group(1)
+        if code == "404":
+            bad(f"Endpoint returned a 404 — the URL is likely wrong or the provider moved it.")
```

Ruff (0.15.12)

[error] f-string without any placeholders

Remove extraneous `f` prefix

(F541)